http.kafka.oneway/templates/all.yaml

```diff
@@ -0,0 +1,135 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: zilla
+  labels:
+    app.kubernetes.io/instance: zilla
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app.kubernetes.io/instance: zilla
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/instance: zilla
+    spec:
+      containers:
+        - name: zilla
+          command: ["/bin/sh", "-c", "/opt/zilla/zilla start -v -e"]
+          image: "ghcr.io/aklivity/zilla:latest"
+          imagePullPolicy: IfNotPresent
+          ports:
+          - containerPort: 8080
+          - containerPort: 9090
+          volumeMounts:
+          - name: config-volume
+            mountPath: /etc/zilla
+          - name: tls-volume
+            mountPath: /etc/zilla/tls
+      volumes:
+      - name: config-volume
+        configMap:
+          name: zilla
+      - name: tls-volume
+        secret:
+          secretName: tls
+          items:
+          - key: localhost.p12
+            path: localhost.p12
```

Comment:
Let's figure out our naming convention for k8s resources.

Initially, my instinct was to _not_ have the type-based suffix, such as `tls-volume` for the `volume` name. But then I looked up the k8s documentation, and found [that approach](https://kubernetes.io/docs/concepts/storage/volumes/#aws-ebs-configuration-example) is used pervasively.

So now I'm wondering, given that these are `zilla-examples`, should we be treating the k8s part as if we are teaching k8s too, with naming suffixes like `...-container`, `...-volume`, etc, or just omit the suffixes everywhere.

What do you think? IMHO we should choose a consistent strategy for all named items regardless.

Reply:
Here's my 2 cents. Generally speaking, I'd avoid type-based suffixes whenever they surface to the user-facing interface. The primary way users access a kubernetes cluster is through `kubectl`. For example, getting some logs:
```
$ kubectl logs --namespace zilla-http-kafka-oneway service/kafka --container zilla
# vs
$ kubectl logs --namespace zilla-http-kafka-oneway-namespace service/kafka-service --container zilla-container

```
I think the first command is really concise where the second one is quite bloated and honestly looks a bit weird.

I regard naming volumes as `something-volume` as a somewhat different case. `Deployment`s in prod can be big, complex yaml structures potentially hard to navigate/comprehend. Specifically, `volume`s (declaration) and `volumeMount`s (usage) are in different places in the yaml tree. Also, if there are multiple containers in the pod that use (some of) the same volumes, the declaration/usage will have a 1:n relationship, so volume info is really spread out to multiple places. IMO having verbose names when it comes to `volume`s is helpful for the developer to navigate a complex Deployment definition. However, this verbosity will not leak to the user-facing interface, because the scope of `volume`/`volumeMount` declarations is really just the `Deployment` where they are declared in.

To sum it up, my take is that omitting type-based suffixes should be the norm, potentially making an exception for cases where verbose names are helpful for the developer but don't result in a bloated/weird user interface. Anyway, given these are simple examples where complexity isn't an issue at this point, we can also decide to completely avoid type suffixes for the sake of a simple/consistent naming scheme.

Reply:
Great discussion points @attilakreiner  I better understand your thought process now.

Per your last point, let's remove the suffixes as this is a simple example, but keep it in mind for more complex k8s configurations.